types/block.go

```diff
@@ -1057,7 +1057,11 @@ func (data *Data) Hash() tmbytes.HexBytes {
 
 	// compute the data availability header
 	// todo(evan): add the non redundant shares back into the header
-	shares, _ := data.ComputeShares()
+	shares, _, err := data.ComputeShares(data.OriginalSquareSize)
+	if err != nil {
+		// todo(evan): see if we can get rid of this panic
+		panic(err)
```

Comment:
I think the only way this can panic is via a bug in nmt. And then it probably should. But we should definitely confirm this.